src/App.tsx

```diff
@@ -1,14 +1,45 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { getPreparedTodos } from './utils/getPreparedTodos';
+import { getTodoById } from './utils/getTodoById';
+import { CompletionQuery } from './types/CompletionQuery';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [completionQuery, setCompletionQuery] = useState<CompletionQuery>(
+    CompletionQuery.All,
+  );
+  const [loading, setLoading] = useState(false);
+
+  const [selectedTodoId, setSelectedTodoId] = useState<number>(0);
+
+  useEffect(() => {
+    setLoading(true);
+
+    getTodos()
+      .then(setTodos)
+      .finally(() => setLoading(false));
```

Comment:
It's better to move the logic inside the `useEffect `hook to a separate function, as it improves code readability, reusability, and makes testing easier. It also helps keep `useEffect `focused on side effects.

Move `useEffect `to the end of a component to improve readability and maintainability.